packages/extension/src/ui/features/actions/ActionScreen.tsx

```diff
@@ -134,20 +134,30 @@ export const ActionScreen: FC = () => {
     case "ALPH_TRANSACTION":
       return (
         <ApproveTransactionScreen
-          transaction={action.payload}
+          transactionParams={action.payload}
           actionHash={action.meta.hash}
-          onSubmit={async (builtTransaction) => {
+          onSubmit={async (builtTransactions) => {
             analytics.track("signedTransaction", {
               networkId: selectedAccount?.networkId || "unknown",
             })
-            if (!builtTransaction || "error" in builtTransaction) {
+
+            if (!builtTransactions || builtTransactions.length === 0) {
               useAppState.setState({
-                error: `${t('Transaction building failed')}: ${builtTransaction?.error || t("unknown")}`,
+                error: `${t('Transaction building failed')}`,
                 isLoading: false,
               })
               navigate(routes.error())
             } else {
-              await approveAction(action, builtTransaction)
+              const errorTransaction = builtTransactions.find(tx => 'error' in tx);
+              if (errorTransaction && 'error' in errorTransaction) {
+                useAppState.setState({
+                  error: `${t('Transaction building failed')}: ${errorTransaction.error}`,
+                  isLoading: false,
+                })
+                navigate(routes.error())
+              }
+
+              await approveAction(action, builtTransactions)
```

Comment:
Shall we wrap these lines of code in an `else` branch?

Reply:
It should be the same without the `else` branch since it would navigate to the error page, but perhaps more clear with the `else` statement. Will fix.